pkg/doc/vc/validate.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package vc
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+)
+
+const (
+	jwtVC = "jwt_vc"
+	ldpVC = "ldp_vc"
+)
+
+/*
+"*/
+
+func ValidateVCFormat(format string) error {
+	if format != jwtVC && format != ldpVC {
+		return fmt.Errorf("unsupported vc format %s, use one of next [%s, %s]", format, jwtVC, ldpVC)
+	}
+	return nil
+}
+
+// SignatureType type of signature used to sign vc.
+type SignatureType string
+
+const (
+	EdDSA  SignatureType = "EdDSA"
+	ES256K SignatureType = "ES256K"
+	ES256  SignatureType = "ES256"
+	ES384  SignatureType = "ES384"
+	PS256  SignatureType = "PS256"
+
+	Ed25519Signature2018        SignatureType = "Ed25519Signature2018"
+	Ed25519Signature2020        SignatureType = "Ed25519Signature2020"
+	EcdsaSecp256k1Signature2019 SignatureType = "EcdsaSecp256k1Signature2019"
+	BbsBlsSignature2020         SignatureType = "BbsBlsSignature2020"
+	JSONWebSignature2020        SignatureType = "JsonWebSignature2020"
+)
+
+// Name of signature type.
+func (st SignatureType) Name() string {
+	return string(st)
+}
+
+// lowerCase name of signature type in lower case. Used internally.
+func (st SignatureType) lowerCase() string {
+	return strings.ToLower(string(st))
+}
+
+func ValidateVCSignatureAlgorithm(format, signatureType string) (SignatureType, error) {
+	var validSignatureTypes []SignatureType
+
+	if format == jwtVC {
+		validSignatureTypes = []SignatureType{
+			EdDSA,
+			ES256K,
+			ES256,
+			ES384,
+			PS256,
+		}
+	}
+
+	if format == ldpVC {
+		validSignatureTypes = []SignatureType{
+			Ed25519Signature2018,
+			Ed25519Signature2020,
+			EcdsaSecp256k1Signature2019,
+			BbsBlsSignature2020,
+			JSONWebSignature2020,
+		}
+	}
+
+	if validSignatureTypes == nil {
+		return "", fmt.Errorf("unsupported vc format %s", format)
+	}
+
+	for _, val := range validSignatureTypes {
+		if val.lowerCase() == strings.ToLower(signatureType) {
+			return val, nil
+		}
+	}
+
```

Comment:
this whole can be replaced by something like this `signatureType.isValid(format)`


Similar to https://github.com/hyperledger/aries-framework-go/blob/main/pkg/wallet/contents.go#L60-L68

Reply:
@vkubiv as discussed, please fix it in next PR